clippy_lints/src/methods/mod.rs

```diff
@@ -3063,3 +3084,15 @@ fn contains_return(expr: &hir::Expr) -> bool {
     visitor.visit_expr(expr);
     visitor.found
 }
+
+fn check_pointer_offset(cx: &LateContext<'_, '_>, expr: &hir::Expr, args: &[hir::Expr]) {
+    if_chain! {
+        if args.len() == 2;
+        if let ty::RawPtr(ty::TypeAndMut { ref ty, .. }) = cx.tables.expr_ty(&args[0]).kind;
+        if let Ok(layout) = cx.tcx.layout_of(ty::ParamEnv::empty().and(ty));
```

Comment:
`TyLayout` has a `is_zst` method. 

Also this panics in dogfood, becacuse `normalize_erasing_regions` fails.

Reply:
There is also [`rustc_lint::types::is_zst`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/types/fn.is_zst.html), which requires a `def_id` for building the `ParamEnv`, before calling `layout_of`:

```rust
fn is_zst<'tcx>(tcx: TyCtxt<'tcx>, did: DefId, ty: Ty<'tcx>) -> bool {
    tcx.layout_of(tcx.param_env(did).and(ty)).map(|layout| layout.is_zst()).unwrap_or(false)
}
```